### PR TITLE
Ova download fix

### PR DESCRIPTION
**Pull request by DaanHoogland**
### Description

This PR allows for downloading OVA with the read properties from OVF flag when the OVF does not contain a minimum hardware version.

Fixes: #6736

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [x] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [x] Minor
- [ ] Trivial

### How Has This Been Tested?

by uploading an OVA that does not provide a minimum hardware version and making sure it is accepted.

### CODE REVIEW

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test centos7 vmware-67u3

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test centos7 vmware-67u3

**Comment by DaanHoogland** [comment]
@borisstoyanov @nvazquez can you please look at this?

**Comment by borisstoyanov** [comment]
@blueorangutan test centos7 vmware-67u3 keepEnv

**Comment by borisstoyanov** [comment]
Sure @DaanHoogland, will do.

File: engine/storage/src/main/java/org/apache/cloudstack/storage/image/deployasis/DeployAsIsHelperImpl.java [hunk 1]
```diff
@@ -190,7 +194,7 @@ private Long retrieveTemplateGuestOsIdFromGuestOsInfo(long templateId, String gu
             }
             return guestOsId;
         } else {
-            throw new CloudRuntimeException("Did not find a guest OS with type " + guestOsType);
+            throw new CloudRuntimeException(String.format("Did not find a guest OS (%s) with type \"%s\" and minimal hyperviser hardware version %s.", guestOsDescription, guestOsType, minimumHardwareVersion));
```
Comment (nvazquez) [inline engine/storage/src/main/java/org/apache/cloudstack/storage/image/deployasis/DeployAsIsHelperImpl.java:197]: Minor typo on 'hyperviser'

Comment (DaanHoogland) [reply engine/storage/src/main/java/org/apache/cloudstack/storage/image/deployasis/DeployAsIsHelperImpl.java:197]:
```suggestion
            throw new CloudRuntimeException(String.format("Did not find a guest OS (%s) with type \"%s\" and minimal hypervisor hardware version %s.", guestOsDescription, guestOsType, minimumHardwareVersion));
```

**Comment by borisstoyanov** [comment]
@blueorangutan test centos7 vmware-67u3 keepEnv

**Review by borisstoyanov — CHANGES_REQUESTED** [review]
Upload from local fails when the user does not select 'read from OVA'
Screenshot 2022-09-30 at 14 34 49

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
> Upload from local fails when the user does not select 'read from OVA' Screenshot 2022-09-30 at 14 34 49

@borisstoyanov I tested this with the macchinina template and it works, than I investigated the template for the cisco PI, it contains four disk definitions and only one vmdk. I am not sure if we should support this type of template, or how to handle it.

**Review by borisstoyanov — APPROVED** [review]
Just had a talk with @DaanHoogland and we agree that this template is not meant to be used without reading the properties, so it does not makes sense to look into that. 
Also run a few upload tests and all seems to be good. LGTM

**Pull request merged**